sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/io/GenericDlqProvider.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.schemas.io;
+
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
+import org.apache.beam.sdk.annotations.Internal;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PDone;
+
+/** A Provider for generic DLQ transforms that handle deserialization failures. */
+@Internal
+@Experimental(Kind.SCHEMAS)
+public interface GenericDlqProvider {
+  /** Returns an id that uniquely represents this Dlq Provider. */
+  String identifier();
+
+  /** Generate a DLQ output from the provided config value. */
+  PTransform<PCollection<Failure>, PDone> newDlqTransform(String config);
```

Comment:
Is there better to use Class as the parameter to look up the implementation, compared to those code to pass a key value pair string?

Reply:
I don't think so. All that would do is push the responsibility on each individual caller to parse the input string into its components, as opposed to centralizing it.  Note that GenericDlq strips the prefix "key" part of this, so all that is passed is the "value" part (the pub/sub topic/ lite topic/ bigquery spec)

Reply:
SG